### PR TITLE
Upgrade to rust2018

### PR DESCRIPTION
**Pull request by faern**
As we discussed on IRC. Let's upgrade all these three crates to Rust 2018 edition and get rid of some legacy conditional compilation etc.

Will greatly aid the work to integrate it into libstd over at #119

### CODE REVIEW

File: core/src/parking_lot.rs [hunk 1]
```diff
@@ -175,14 +168,9 @@ where
 {
     // Try to read from thread-local storage, but return None if the TLS has
     // already been destroyed.
-    #[cfg(has_localkey_try_with)]
     fn try_get_tls(key: &'static LocalKey<ThreadData>) -> Option<*const ThreadData> {
         key.try_with(|x| x as *const ThreadData).ok()
     }
```
Comment (Amanieu) [inline core/src/parking_lot.rs]: There isn't much point keeping that as a separate function now.

File: src/condvar.rs [hunk 2]
```diff
@@ -668,7 +670,7 @@ mod tests {
         let mut g = m.lock();
         while !c.notify_one() {
             // Wait for the thread to get into wait()
-            ::MutexGuard::bump(&mut g);
+            crate::MutexGuard::bump(&mut g);
```
Comment (Amanieu) [inline src/condvar.rs]: Just add `MutexGuard` to the import above.

File: src/once.rs [hunk 3]
```diff
@@ -16,8 +16,8 @@ use std::sync::atomic::AtomicUsize as AtomicU8;
 type U8 = usize;
 use parking_lot_core::{self, SpinWait, DEFAULT_PARK_TOKEN, DEFAULT_UNPARK_TOKEN};
 use std::fmt;
```
Comment (Amanieu) [inline src/once.rs]: `core::fmt`

File: core/src/parking_lot.rs [hunk 4]
```diff
@@ -5,19 +5,20 @@
...
-use std::time::{Duration, Instant};
-use thread_parker::ThreadParker;
-use util::UncheckedOptionExt;
-use word_lock::WordLock;
+use core::{
+    cell::{Cell, UnsafeCell},
+    ptr,
+};
+use std::{
+    sync::atomic::{AtomicPtr, AtomicUsize, Ordering},
```
Comment (Amanieu) [inline core/src/parking_lot.rs]: `sync::atomic` should be imported from `core`, not `std`.

File: core/src/parking_lot.rs (hunk 1 above)
Comment (faern) [reply core/src/parking_lot.rs]: True. Removed.

**Pull request merged**